zirgen/Dialect/BigInt/Precompiles.md

```diff
@@ -0,0 +1,106 @@
+# Creating a precompile using the BigInt2 system
+
+Creating a new accelerator using BigInt2 requires a series of related changes across files in two repositories. This is the process you should follow if you want to create a new bigint2 precompile and make it available within the zkVM.
+
+To keep the demo simple, let's imagine that we want to create a trivial accelerator which sums two 128-bit numbers, which we'll call "add128".
+
+
+First, create files under [`zirgen/circuit/bigint/`] which will hold the accelerator builder function. We might call them `add128.h` and `add128.cpp`, declaring and then defining a function like this:
+
+```
+void genAdd128(mlir::OpBuilder& builder, mlir::Location loc) {
+    auto lhs = builder.create<BigInt::LoadOp>(loc, 128, 11, 0);
+    auto rhs = builder.create<BigInt::LoadOp>(loc, 128, 12, 0);
+    auto sum = builder.create<BigInt::AddOp>(loc, lhs, rhs);  
+    builder.create<BigInt::StoreOp>(loc, sum, 13, 0);
+}
+```
+
+We might also like to create a test function here, but we'll gloss over that for now.
+
+To make use of the new function, we must add it as an option of the `bigint2c` compiler program, which generates the precompiled blobs we'll need to load into the zkVM. Inside `zirgen/circuit/bigint/bigint2c.cpp`, we'll add a new command line option:
+
+```
+enum class Program {
+  ModPow65537,
+  EC_Double,
+  EC_Add,
+  Add128
+};
+```
+
+Make this option available by extending the arg parser's declaration:
+```
+   
+               clEnumValN(Program::Add128, "add128", "Add128")),
+```
+
+Later, in the main function, we must handle the new option by invoking our new generator function:
+
+```
+case Program::Add128:
+  zirgen::BigInt::genAdd128(builder, loc);
+  break;
+```
+
```

Comment:
I think bootstrapping is a separate section from bigint blob compilation; that section describes how to add the ability to compile a single precompile, whereas the bootstrapping section is about adding the new precompile to the list of all precompiles to be built when by our bootstrapping command (i.e., our "compile all our precompiles and copy them to the `risc0` repo" command)

```suggestion

## Bootstrapping your precompile into the `risc0` repo

```